src/lib.rs

```diff
@@ -239,14 +258,28 @@ fn setup_bitcoind(
         log::info!("Creating a new watchonly wallet on bitcoind.");
         bitcoind.create_watchonly_wallet(&config.main_descriptor)?;
         log::info!("Watchonly wallet created.");
-    } else {
+    } else if !is_remote {
         #[cfg(windows)]
         if !cfg!(test) && !wo_path.exists() {
             return Err(StartupError::NoWatchonlyInDatadir);
         }
     }
     log::info!("Loading our watchonly wallet on bitcoind.");
-    bitcoind.maybe_load_watchonly_wallet()?;
+    if is_remote {
+        if let Err(e) = bitcoind.maybe_load_watchonly_wallet() {
+            let wallets = bitcoind.list_wallets();
+            // if a wallet w/ legacy name exist
+            if wallets.contains(&wo_path_str) && !wallets.contains(&new_wo_path_str) {
```

Comment:
To avoid repeating logic, could you call `maybe_load_watchonly_wallet()` directly without performing this check here?

Perhaps you can check instead if the first attempt failed because the wallet doesn't exist by checking the error code, which I think would be `-18`: https://github.com/bitcoin/bitcoin/blob/41544b8f96dbc9c6b8998acd6522200d67cdc16d/src/rpc/protocol.h#L81.